### ISSUE TITLE
Talos v1.5.0 failing to install

### ISSUE THREAD

**Issue by EternalFr0st** [siderolabs/talos#7626]
## Bug Report
Talos v1.5.0 failing to install to seemingly running out of space in the temporary root while booted from the ISO

### Description

Hello, trying to install 1.5.0 on a Proxmox VM.

Using the metal-amd64.iso downloaded from releases page and the VM has 4 core and 16 GB of ram.
The control plane config has been modified minimally, just added the static network configuration and custom CA certificate.
However when applying the config, the VM tries to pull the installer and fails to extract it due to running out of space on device, which I assume is the temporary live ISO root.
Not entirely sure how that happens, since I assume it just runs off of RAM and there's plenty...
Let me know if you need any more information and appreciate any help or information!

Adding screenshot of logs below

### Logs

![image](https://github.com/siderolabs/talos/assets/87126464/c049aefb-b1cd-40d2-bfc8-924768627953)

### Environment

- Talos version: 1.5.0
- Kubernetes version: [`kubectl version --short`]
- Platform: Proxmox 8.0.4

**Comment by agmtr** [comment]
Also ova image problem - talos-cp-1/talos-cp-1.vmdk is larger than the maximum size supported by datastore

**Comment by smira** [comment]
> Also ova image problem - talos-cp-1/talos-cp-1.vmdk is larger than the maximum size supported by datastore

Let's keep a separate issue for it, and this can be fixed easily by shrinking the image size.

**Comment by smira** [comment]
@EternalFr0st do you have full boot logs to verify RAM size?

**Comment by EternalFr0st** [comment]
@smira Can't pull the logs, since it's ISO booted, unable to do dmesg or logs from talosctl, but after switching to the monitor tab with F3 in the console, I did notice it's limited to ~1GB, screenshot below.

![image](https://github.com/siderolabs/talos/assets/87126464/2d4efd4b-66cb-4e1d-85a9-fb834b8edf25)

After fiddling with VM settings, I narrowed it down to enabling RAM Hotplug functionality in Proxmox ( NUMA enabled, without RAM Hotplug is fine ) is what limited the RAM to 1G, once disabled the full RAM amount shows up.

While RAM hotplug is just something I enable by default and I'm fine with disabling it, it's probably not supported by Talos at all, I would be curious to hear if there's a kernel option I could enable to test it out "for fun"? Or just any information as to why this would be happening?

Other than that, I think we're good to close, sorry for the troubles :)
